src/main/java/uk/gov/hmcts/ccd/domain/service/FindUserProfileOperation.java

```diff
@@ -18,9 +28,35 @@ public FindUserProfileOperation(UserProfileRepository userProfileRepository) {
     }
 
     public UserProfile execute(String userProfileId, String actionedBy) {
-        final UserProfile userProfile = userProfileRepository.findById(userProfileId, actionedBy);
+        String decodedUid = UriUtils.decode(userProfileId, "UTF-8");
+        final UserProfile userProfile = userProfileRepository.findById(decodedUid, actionedBy);
         return Optional.ofNullable(userProfile)
             .orElseThrow(() -> new NotFoundException("Cannot find user profile"));
     }
+
+    public UserProfileCollection execute(List<String> userProfileIds, String actionedBy) {
+        validateRequest(userProfileIds);
+        final List<UserProfile> userProfiles = userProfileIds.stream()
```

Comment:
this will be causing N+1 problem if the operation is used with multiple emails later on.

Reply:
Done